configure.ac

```diff
@@ -2226,12 +2244,23 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 ## NOTE: clock_gettime may require linking to the rt or posix4 library
 ##       so we'll search for it before calling AC_CHECK_FUNCS.
 AC_SEARCH_LIBS([clock_gettime], [rt posix4])
-AC_CHECK_FUNCS([alarm asprintf clock_gettime fcntl flock fork])
+AC_CHECK_FUNCS([asprintf clock_gettime fcntl flock fork])
 AC_CHECK_FUNCS([gethostname getrusage gettimeofday])
 AC_CHECK_FUNCS([rand_r random])
 AC_CHECK_FUNCS([strcasestr strdup symlink])
 AC_CHECK_FUNCS([tmpfile vasprintf waitpid])
 
+case "$host_os" in
+  *mingw*)
+    # alarm(2) support is spotty in MinGW, so assume it doesn't exist
+    #
+    # https://lists.gnu.org/archive/html/bug-gnulib/2013-03/msg00040.html
+    ;;
+  *)
+    AC_CHECK_FUNCS([alarm])
+    ;;
+esac
```

Comment:
Alarm is flaky on MinGW and sometimes gets picked up via AC_CHECK_FUNCS, even when it doesn't work.